uber/site_sections/budget.py

```diff
@@ -46,3 +46,157 @@ def mpoints(self, session):
         all = [(sum(mpu.amount for mpu in mpus), group, mpus)
                for group, mpus in groups.items()]
         return {'all': sorted(all, reverse=True)}
+
+    @ajax
+    def add_promo_code_words(self, session, text='', part_of_speech=0):
```

Comment:
Oh sorry, I don't mean doing this as opposed to having promo_code_stubs in config -- it's definitely better this way. I meant doing promo code stubs at _all_. We don't -- or I don't -- know how many admins will actually use auto-generated promo codes as opposed to making them manually.